_includes/pbc/all/install-features/202400.0/install-the-packaging-units-feature.md

```diff
@@ -0,0 +1,1138 @@
+
+class PickingListDependencyProvider extends SprykerPickingListDependencyProvider
+{
+    /**
+     * @return list<\Spryker\Zed\PickingListExtension\Dependency\Plugin\PickingListCollectionExpanderPluginInterface>
+     */
+    protected function getPickingListCollectionExpanderPlugins(): array
+    {
+        return [
+            new ProductPackagingUnitPickingListCollectionExpanderPlugin(),
+        ];
+    }
+}
+```
+
+**src/Pyz/Glue/PickingListsBackendApi/PickingListsBackendApiDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Glue\PickingListsBackendApi;
+
+use Spryker\Glue\PickingListsBackendApi\PickingListsBackendApiDependencyProvider as SprykerPickingListsBackendApiDependencyProvider;
+use Spryker\Glue\ProductPackagingUnitsBackendApi\Plugin\PickingListsBackendApi\ProductPackagingUnitPickingListItemsBackendApiAttributesMapperPlugin;
+
+class PickingListsBackendApiDependencyProvider extends SprykerPickingListsBackendApiDependencyProvider
+{
+    /**
+     * @return list<\Spryker\Glue\PickingListsBackendApiExtension\Dependency\Plugin\PickingListItemsBackendApiAttributesMapperPluginInterface>
+     */
+    protected function getPickingListItemsBackendApiAttributesMapperPlugins(): array
+    {
+        return [
+            new ProductPackagingUnitPickingListItemsBackendApiAttributesMapperPlugin(),
+        ];
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Add an item with packaging units to cart and check if the following statements are true:
+
+* A packaging unit can be found for an item.
+* The `amount`, `amountSalesUnit`, `amountLeadProduct` and `ProductPackagingUnit` fields in the `ItemTransfer` properties get fully populated.
+* The amount restriction works as expected.
+* Availability is validated respectfully according to your lead product's and packaging unit's configuration.
+* Item grouping in the cart works as expected.
+* Variable amount changes affect unit prices in the `ItemTransfer` properties.
+* The quantity and amount are merged correctly when the group key matches.
+
+Go through the checkout workflow, make an order and check if the following statements are true:
+* Check if the stock is modified respectfully according to your lead product's and packaging unit's configuration.
+* Check if the following fields in the `spy_sales_order_item` table are saved:
+  * `amount`
+  * `amount_sku`
+  * `amount_measurement_unit_name`
+  * `amount_measurement_unit_code`
+  * `amount_measurement_unit_precision`
+  * `amount_measurement_unit_conversion`
+  * `amount_base_measurement_unit_name`
+
+Go to the Zed UI Sales overview, check the order, and verify the following:
+- The correct sales unit is displayed.
+- The correct amount is displayed per sales order item.
+
+Make sure the following:
+* Abstract products that have packaging units available don’t have `add_to_cart_sku` field at Elasticsearch document.
```

Comment:
"Elasticsearch document" seems wrong here.

Reply:
Could you specify please what exactly is wrong?
We use ElasticSearch as a search engine.
I checked the case when the product has no packaging units: there is an `add_to_cart_sku` field in the elastic search document. But the product which has packaging units has no such field. That's what is mentioned here.